staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -523,7 +523,7 @@ func (r *claimResolver) resolve(endpoint endpoint, allClaims claims) error {
 	return nil
 }
 
-func (a *Authenticator) AuthenticateToken(token string) (user.Info, bool, error) {
+func (a *Authenticator) AuthenticateToken(aud []string, token string) (user.Info, bool, error) {
```

Comment:
our oidc authenticator already checks audience based on the `--oidc-client-id`. should `--oidc-client-id` be added to the kube-apiserver's `AuthenticationInfo.APIAudiences`? should this method short-circuit if `aud` is populated and does not contain ClientID?